core/src/main/java/org/pentaho/di/core/row/ValueDataUtil.java

```diff
@@ -0,0 +1,2255 @@
+      if ( !isSpace( str.charAt( i ) ) ) {
+        return false;
+      }
+    }
+    return true;
+  }
+
+  /**
+   * Checks an xml file is well formed.
+   *
+   * @param metaA
+   *          The ValueMetaInterface
+   * @param dataA
+   *          The value (filename)
+   * @return true if the file is well formed.
+   * @deprecated Use {@link ValueDataUtil#isXMLFileWellFormed(ValueMetaInterface, Object, boolean)} instead
+   */
+  @Deprecated
+  public static boolean isXMLFileWellFormed( ValueMetaInterface metaA, Object dataA ) {
+    boolean xmlWellFormed = false;
+    try {
+      xmlWellFormed = isXMLFileWellFormed( metaA, dataA, false );
+    } catch ( KettleFileNotFoundException e ) {
+      // Ignore
+    }
+    return xmlWellFormed;
+  }
+
+  /**
+   * Checks an xml file is well formed.
+   *
+   * @param metaA
+   *          The ValueMetaInterface
+   * @param dataA
+   *          The value (filename)
+   * @param failIfNoFile
+   *          Indicates if the transformation should fail if no file is found
+   * @return true if the file is well formed.
+   * @throws KettleFileNotFoundException
+   */
+  public static boolean isXMLFileWellFormed( ValueMetaInterface metaA, Object dataA, boolean failIfNoFile )
+          throws KettleFileNotFoundException {
+    if ( dataA == null ) {
+      return false;
+    }
+
+    String filename = dataA.toString();
+    FileObject file = null;
+    try {
+      file = KettleVFS.getFileObject( filename );
+      throwsErrorOnFileNotFound( file );
+      return XMLCheck.isXMLFileWellFormed( file );
+    } catch ( KettleFileNotFoundException e ) {
+      if ( failIfNoFile ) {
+        throw e;
+      }
+      log.debug( e.getMessage() );
+    } catch ( Exception e ) {
+      log.debug( e.getMessage() );
+    } finally {
+      IOUtils.closeQuietly( file );
+    }
+    return false;
+  }
+
+  /**
+   * Checks an xml string is well formed.
+   *
+   * @param metaA
+   *          The ValueMetaInterface
+   * @param dataA
+   *          The value (filename)
+   * @return true if the file is well formed.
+   */
+  public static boolean isXMLWellFormed( ValueMetaInterface metaA, Object dataA ) {
+    if ( dataA == null ) {
+      return false;
+    }
+    try {
+      return XMLCheck.isXMLWellFormed( new ByteArrayInputStream( metaA.getBinary( dataA ) ) );
+    } catch ( Exception e ) {
+      log.debug( e.getMessage() );
+    }
+    return false;
+  }
+
+  /**
+   * Get file encoding.
+   *
+   * @param metaA
+   *          The ValueMetaInterface
+   * @param dataA
+   *          The value (filename)
+   * @return file encoding.
+   * @deprecated Use {@link ValueDataUtil#getFileEncoding(ValueMetaInterface, Object, boolean)} instead
+   */
+  @Deprecated
+  public static String getFileEncoding( ValueMetaInterface metaA, Object dataA ) throws KettleValueException {
+    String encoding = null;
+    try {
+      encoding = getFileEncoding( metaA, dataA, true );
+    } catch ( KettleFileNotFoundException e ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)